### PR TITLE
Improve model for thickeners

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This PR adds an improved, predictive model for thickener/clarifiers which replaces the previous thickener model.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/models/unit_models/solid_liquid/tests/test_thickener.py [hunk 1]
```diff
@@ -39,44 +38,43 @@
...
         Callable method for Block construction.
         """
         super().build()
 
-        self._state_block_class = TestStateBlock
+        self._state_block_class = SolidStateBlock
 
         # Add Phase objects
-        self.Liq = LiquidPhase()
+        self.sol = SolidPhase()
```
Comment (MarcusHolly) [inline idaes/models/unit_models/solid_liquid/tests/test_thickener.py:77]:
```suggestion
        self.Sol = SolidPhase()
```
Minor comment, but isn't this usually capitalized?

File: idaes/models/unit_models/solid_liquid/tests/test_thickener.py [hunk 2]
```diff
@@ -105,6 +103,107 @@ def build(self):
...
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super().build()
+
+        self._state_block_class = LiquidStateBlock
+
+        # Add Phase objects
+        self.liq = LiquidPhase()
```
Comment (MarcusHolly) [inline idaes/models/unit_models/solid_liquid/tests/test_thickener.py:174]: Same here

File: idaes/models/unit_models/solid_liquid/tests/test_thickener.py (hunk 1 above)
Comment (andrewlee94) [reply idaes/models/unit_models/solid_liquid/tests/test_thickener.py:77]: It is a name, so it doesn't really matter (plus it is just for a test).

File: idaes/models/unit_models/solid_liquid/thickener.py [hunk 3]
```diff
@@ -66,110 +214,334 @@ def build(self):
...
+        # Setup StateBlock argument dict
+        tmp_dict = dict(**self.config.solid_property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["defined_state"] = True
+
+        self.solid_inlet_state = self.config.solid_property_package.build_state_block(
+            self.flowsheet().time, doc="Solid properties in separator", **tmp_dict
+        )
+
+        # Add solid spliter
```
Comment (MarcusHolly) [inline idaes/models/unit_models/solid_liquid/thickener.py:227]:
```suggestion
        # Add solid splitter
```

File: idaes/models/unit_models/solid_liquid/thickener.py [hunk 4]
```diff
@@ -66,110 +214,334 @@ def build(self):
...
+        # Setup StateBlock argument dict
+        tmp_dict = dict(**self.config.liquid_property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["defined_state"] = True
+
+        self.liquid_inlet_state = self.config.liquid_property_package.build_state_block(
+            self.flowsheet().time, doc="liquid properties in separator", **tmp_dict
+        )
+
+        # Add liquid spliter
```
Comment (MarcusHolly) [inline idaes/models/unit_models/solid_liquid/thickener.py:260]:
```suggestion
        # Add liquid splitter
```

File: idaes/models/unit_models/solid_liquid/thickener.py [hunk 5]
```diff
@@ -66,110 +214,334 @@ def build(self):
...
+        @self.Constraint(self.flowsheet().time)
+        def underflow_flowrate(b, t):
+            return b.flow_vol_underflow[t] == (
+                b.solid_split.underflow_state[t].flow_vol
+                + units.convert(
+                    b.liquid_split.underflow_state[t].flow_vol, to_units=uom.FLOW_VOL
+                )
+            )
+
+        # 2.8
```
Comment (MarcusHolly) [inline idaes/models/unit_models/solid_liquid/thickener.py:410]: What;s 2.8 mean here?

**Review by MarcusHolly — APPROVED** [review]
LGTM - just a few small comments

**Review by agarciadiego — APPROVED** [review]
LGTM

**Pull request merged**